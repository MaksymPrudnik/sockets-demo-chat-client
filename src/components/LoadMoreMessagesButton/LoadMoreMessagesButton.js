import React from 'react';
import { socket } from '../../service/socket';

const LoadMoreMessagesButton = ({count, channel}) => {
    const handleLoadMore = () => {
        console.log(count)
        socket.emit('load messages', {count, channel});
    }
    return <button 
        style={{border: 'none',
        backgroundColor: 'transparent', 
        textDecoration: 'underline'}}
        onClick={handleLoadMore}
     >Load more</button>
}

export default LoadMoreMessagesButton;