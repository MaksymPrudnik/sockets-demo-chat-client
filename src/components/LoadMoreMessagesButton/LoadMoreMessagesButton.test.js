import React from 'react';
import { shallow } from 'enzyme';
import LoadMoreMessagesButton from './LoadMoreMessagesButton';

describe('<LoadMoreMessagesButton />', () => {
    const props = {channel: 'new channel', count: 5}
    const wrapper = shallow(<LoadMoreMessagesButton {...props} />);
    
    it('should match the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })
})