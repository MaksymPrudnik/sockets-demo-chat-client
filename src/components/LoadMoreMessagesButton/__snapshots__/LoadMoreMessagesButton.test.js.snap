// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadMoreMessagesButton /> should match the snapshot 1`] = `"<button style=\\"border:none;background-color:transparent;text-decoration:underline\\">Load more</button>"`;
